{
  "WorkItem": {
    "AffectedComponent": {
      "Name": "",
      "DisplayName": ""
    },
    "ClosedComment": "This should be fixed in our new release, 0.8.2.0. Please give it a try and reactive if you're still seeing this problem. Thanks again for reporting it.",
    "ClosedDate": "2013-01-21T18:40:38.107-08:00",
    "CommentCount": 1,
    "Custom": null,
    "Description": "I created a module, and inside the module added two classes, one private, the other public. The public class has a private member, that is an array of the first (private) class. Like so:\n\ntest.ts\n-----------------------\nmodule Test {\n    class A {\n        one: string;\n        two: bool;\n        constructor (t: string) {\n            this.one = t;\n            this.two = false;\n        }\n    }\n    export class B {\n        private member: A[];\n\n        constructor () {\n            this.member = [];\n        }\n    }\n}\n-----------------------\n\nIf I now add another file, which has a module with the same name, I get an error message on the private member declaration, that says A is not recognized. Here is the other file:\n\ntest2.ts\n-----------------------\nmodule Test {\n}\n-----------------------\n\nAnd here is the command line:\n\ntsc test2.ts test.ts\n\nAnd the error:\n\nC:/.../test.ts(11,24): The name 'A' does not exist in the current scope\n\nThe error only happens, if the test2.ts is before test.ts on the command line. \nOtherwise it does not happen.\n\nThe error does not happen if I compile test.ts by itself.\n\nThe error can be made to go away by using any[] as the type of the member, which of course is bad because type information is lost.\n\nIf I export class A, the error goes away. But that is also bad because I want to hide it.\n\nIf I change the module name in test2.ts, the error goes away.\n\nFor some reason A is excluded from the available symbols in module Test if the module is defined in more than one file.",
    "LastUpdatedDate": "2014-01-10T10:34:33.287-08:00",
    "PlannedForRelease": "",
    "ReleaseVisibleToPublic": false,
    "Priority": {
      "Name": "Low",
      "Severity": 50,
      "Id": 1
    },
    "ProjectName": "typescript",
    "ReportedDate": "2012-11-20T06:40:55.283-08:00",
    "Status": {
      "Name": "Closed",
      "Id": 4
    },
    "ReasonClosed": {
      "Name": "Unassigned"
    },
    "Summary": "Private class reference causes compiler error",
    "Type": {
      "Name": "Issue",
      "Id": 3
    },
    "VoteCount": 1,
    "Id": 456
  },
  "FileAttachments": [],
  "Comments": [
    {
      "Message": "Thanks for reporting this issue! Marking 'active' to be fixed. We shouldn't be depending on the order of the files on the commandline.",
      "PostedDate": "2012-11-26T14:47:20.327-08:00",
      "Id": 83059
    }
  ]
}